### DISCUSSION TITLE
RAX3000M NAND 从 ImmortalWrt 到 OpenWrt

### DISCUSSION THREAD

**Discussion by cherinyy** [immortalwrt/immortalwrt#1128] [Q&A] [answered]
我想把 RAX3000M NAND 从 ImmortalWrt 换到 OpenWrt 试用一下。

根据  https://github.com/openwrt/openwrt/commit/423186d7d8b4f23aee91fca4f1774a195eba00d8 上的教程，在刷入 `BL2` 时遭遇了问题：

```
Could not open mtd device: BL2
Can't open device for writing!
```

我现在应该知道问题出现在哪里了，没有解锁读写权限，需要安装 `kmod-mtd-rw` 然后执行 `insmod mtd-rw.ko i_want_a_brick=1`。
但等我看到这个的时候，已经刷完了。

想问一下这个有关系吗，我已经成功刷入 `FIP`，通过 TFTP 服务器刷入 initramfs 映像，然后调整布局，最后刷入了 Sysupgrade 映像。

那么我在 OpenWrt 下，还需要继续刷入 `bl2` 吗？

**Comment by 1715173329** [comment, marked as answer]
最好是刷一下。

**Comment by wenewzhang** [comment]
我就是没有刷BL2, 以为有一个FIP就可以了,结果变砖了,现在能ping通路由器,但无法刷机,也进不了原来的 ImmortalWrt系统

**Comment by yiquanfeng** [comment]
有没有mtd-rw.ko的下载地址

**Reply by gentrydeng** [reply]
> 有没有mtd-rw.ko的下载地址

```
opkg update && opkg install kmod-mtd-rw && insmod mtd-rw.ko i_want_a_brick=1
```

**Reply by yiquanfeng** [reply]
其实我提这个问题的时候，我的路由器是没有网络的

**Reply by yiquanfeng** [reply]
不过我后面解决了联网的问题，也感谢你了
